pinot-core/src/main/java/com/linkedin/pinot/core/plan/FilterPlanNode.java

```diff
@@ -52,10 +50,12 @@
   private static final Logger LOGGER = LoggerFactory.getLogger(FilterPlanNode.class);
   private final BrokerRequest _brokerRequest;
   private final IndexSegment _segment;
+  private boolean _optimizeAlwaysFalse;
 
   public FilterPlanNode(IndexSegment segment, BrokerRequest brokerRequest) {
     _segment = segment;
     _brokerRequest = brokerRequest;
+    _optimizeAlwaysFalse = true;
```

Comment:
This is for unit test, which creates a plan node with and without this optimization and ensures that all docIds gotten from the filter operators are identical.